modules/core/include/opencv2/core/fast_math.hpp

```diff
@@ -130,18 +136,8 @@ cvRound( double value )
  */
 CV_INLINE int cvFloor( double value )
 {
-#if (defined _MSC_VER && defined _M_X64 || (defined __GNUC__ && defined __SSE2__ && !defined __APPLE__)) && !defined(__CUDACC__)
```

Comment:
1) Modern compilers is able to apply more effective construction instead of proposed intrinsic code (`comiss+set<cond>` instead of `movmskps`).
2) In general, intrinsic code may block some compiler optimizations including loop unrolling and auto-vectorizer.
3) There is bug in some version of MSVC compiler with AVX code generation mode: it omits `_mm_set_*` statement ("value" is already stored as XMM register), so higher elements are not cleared to zero. As a result, "movemask" returns values in range 0-15, that is wrong (0-1 is expected). Related to [this code](https://github.com/opencv/opencv/blob/3.2.0/modules/video/src/lkpyramid.cpp#L466-L467).

There are MSVS2015 times for "floor" processing of 1M float vector on i5-6300U:
```
Current: (samples = 100, mean = 1.46, median = 1.38, stddev = 0.20 (13.8%))
Proposed: (samples = 100, mean = 0.78, median = 0.79, stddev = 0.12 (15.2%))
```
BTW, There are similar results for SSE4.2 and AVX2 compilation flags

GCC on i5-6600:
```
Current: (samples = 10, mean = 5.82, median = 5.75, stddev = 0.15 (2.5%))
Proposed: (samples = 100, mean = 0.38, median = 0.37, stddev = 0.05 (13.8%))
```